x-pack/plugins/alerting/server/task_runner/transform_action_params.ts

```diff
@@ -89,3 +98,48 @@ export function transformActionParams({
     variables
   );
 }
+
+export function transformSummaryActionParams({
+  alerts,
+  rule,
+  ruleTypeId,
+  actionsPlugin,
+  actionId,
+  actionTypeId,
+  spaceId,
+  actionParams,
+  ruleUrl,
+  kibanaBaseUrl,
+}: {
+  alerts: SummarizedAlertsWithAll;
+  rule: SanitizedRule<RuleTypeParams>;
+  ruleTypeId: string;
+  actionsPlugin: ActionsPluginStartContract;
+  actionId: string;
+  actionTypeId: string;
+  spaceId: string;
+  actionParams: RuleActionParams;
+  kibanaBaseUrl?: string;
+  ruleUrl?: string;
+}): RuleActionParams {
+  const variables = {
+    kibanaBaseUrl,
+    date: new Date().toISOString(),
+    rule: {
+      params: rule.params,
+      id: rule.id,
+      name: rule.name,
+      type: ruleTypeId,
+      url: ruleUrl,
+      tags: rule.tags,
+      spaceId,
+    },
+    alerts,
```

Comment:
The data structure is a bit misaligned from the functional specs => https://github.com/elastic/kibana/issues/143200. For example, `alerts.new.alerts` should be `alerts.new.data`, same for the other fields.

Reply:
yes, i left them as they are intentionally because i was not sure as the existing getSummaryAlerts method returns them as `new.alerts`... fixing.

Reply:
Done in [#3d137a1](https://github.com/elastic/kibana/pull/147360/commits/3d137a13893d66c7b29ac93e20446f4fefbba12c)